tests/javapc/src/test/scala/hover/HoverDefnSuite.scala

```diff
@@ -0,0 +1,166 @@
+package hover
+
+import tests.pc.BaseJavaHoverSuite
+import java.net.URI
+
+class HoverDefnSuite extends BaseJavaHoverSuite {
+  check(
+    "int value",
+    """
+      |class A {
+      |    public static int NUMBER = 42;
+      |
+      |    public static void main(String args[]){
+      |        int x = NU@@MBER;
+      |    }
+      |}
+      |""".stripMargin,
+    """
+      |public static int NUMBER
+      |""".stripMargin.javaHover,
+  )
+
+  check(
+    "test.clazz",
+    """
+      |import java.util.List;
+      |
+      |class A {
+      |  public static void main(String args[]){
+      |    List<Integer> x = List.of(1);
+      |    @@x;
+      |  }
+      |}
+      |""".stripMargin,
+    """|java.util.List<java.lang.Integer> x
+       |""".stripMargin.javaHover,
+  )
+
+  check(
+    "var assigment",
+    """
+      |import java.util.List;
+      |
+      |class A {
+      |  public static void main(String args[]){
+      |    var @@x = List.of(1);
+      |  }
+      |}
+      |""".stripMargin,
+    """|java.util.List<java.lang.Integer> x
+       |""".stripMargin.javaHover,
+  )
+
+  check(
+    "class definition",
+    """
+      |public class @@A {
+      |  public static void main(String args[]){
+      |  }
+      |}
+      |""".stripMargin,
+    "public class class_definition.A".javaHover,
+  )
+
+  check(
+    "class definition with inheritance",
+    """
+      |interface Foo {}
+      |
+      |interface Bar {}
+      |
+      |abstract class Baz {}
+      |
+      |public class @@A extends Baz implements Foo, Bar {
+      |  public static void main(String args[]){
+      |  }
+      |}
+      |""".stripMargin,
+    "public class class_definition_with_inheritance.A extends class_definition_with_inheritance.Baz implements class_definition_with_inheritance.Foo, class_definition_with_inheritance.Bar".javaHover,
```

Comment:
Potentially as a next step we would need to shorten it, but we don't need to do it right away. Let's just sum up anything that needs further work in the separate issue.